lib/src/components/NxFileUpload/__tests__/NxFileUpload.test.tsx

```diff
@@ -38,14 +40,25 @@ describe('NxFileUpload', function() {
   });
 
   it('shows an error when there is no file selected, if it isRequired is not isPristine', function() {
-    expect(render().queryByRole('alert')).not.toBeTruthy();
-    expect(render({ isPristine: true }).queryByRole('alert')).not.toBeTruthy();
-    expect(render({ isRequired: true, isPristine: true }).queryByRole('alert')).not.toBeTruthy();
+    expect(quickRender().queryByRole('alert')).not.toBeTruthy();
+    expect(quickRender({ isPristine: true }).queryByRole('alert')).not.toBeTruthy();
+    expect(quickRender({ isRequired: true, isPristine: true }).queryByRole('alert')).not.toBeTruthy();
 
-    expect(render({ isRequired: true, isPristine: false }).queryByRole('alert'))
+    expect(quickRender({ isRequired: true, isPristine: false }).queryByRole('alert'))
         .toHaveTextContent('This field is Required!');
   });
 
+  it('shows an error when in a form with showValidationErrors if it is isRequired and no file is selected', function() {
+    const renderWithForm = render(
+      <NxForm onSubmit={() => {}} showValidationErrors={true}>
+        <NxFileUpload { ...minimalProps } isRequired={true} />
+      </NxForm>
+    );
+
+    expect(renderWithForm.queryByRole('alert')).toBeTruthy();
+    expect(renderWithForm.queryByRole('alert')).toHaveTextContent('This field is Required!');
```

Comment:
Also check `aria-invalid` and `aria-errormessage`

Reply:
Updated here 45409a16cdfa84a6936a486b465972c6da8d9f39.